x-team/auto-ui
# Change the method to compare values

**Release Type:** *Bug fix* 


Fixes [XP-2159](https://x-team-internal.atlassian.net/browse/XP-2159)

## Description

Apart from plain values like string and numbers, we're also using objects as values. For that reason, the approach to compare if the **editValue** is equal to the field **value** needs to be changed. Now I installed lodash.isEqual which automatically will compare two values, no matter if they are **string**, **number**, **boolean**, **null**, **undefined** or **objects** with one or more properties.

## Checklist

**Before submitting a pull request,** please make sure the following is done:



## Related PRs

List related PRs against other Auto repos if applicable:

branch | PR
------ | ------
xp-2159-allow-editing-fields-master | [Auto](https://github.com/x-team/auto/pull/1530)

## Steps to Test or Reproduce

1. Activate the edition mode by clicking the pencil icon.
2. The **SAVE** button should be disabled when the field value and the current editing value are equal.

## Screenshots
![inline editor](https://user-images.githubusercontent.com/3811134/45859545-e9007000-bd30-11e8-99c9-1484e135c7e2.gif)
